### ISSUE TITLE
'prompt' doesn't accept non-ASCII characters

### ISSUE THREAD

**Issue by L-B** [ansible/ansible#30265]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

vars_prompt

##### ANSIBLE VERSION

```
$ ansible --version
ansible 2.3.2.0
  config file = /work/ansible/configuration/ansible.cfg
  configured module search path = Default w/o overrides
  python version = 2.7.5 (default, Nov  6 2016, 00:28:07) [GCC 4.8.5 20150623 (Red Hat 4.8.5-11)]
```

##### CONFIGURATION

N/A

##### OS / ENVIRONMENT

CentOS 7

##### SUMMARY

Non-ASCII characters in the prompt of a vars_prompt section cause errors. This looks very much like #10319, however, it was closed a long time ago, and this is happening on a recent release.

##### STEPS TO REPRODUCE

```yaml
  vars_prompt:
    - name: "test"
      prompt: "é"

```

##### EXPECTED RESULTS

It should prompt for tha variable:
```
é:
```

##### ACTUAL RESULTS

```
ERROR! Unexpected Exception: 'ascii' codec can't encode character u'\xe9' in position 0: ordinal not in range(128)
the full traceback was:

Traceback (most recent call last):
  File "/usr/bin/ansible-playbook", line 109, in <module>
    exit_code = cli.run()
  File "/usr/lib/python2.7/site-packages/ansible/cli/playbook.py", line 154, in run
    results = pbex.run()
  File "/usr/lib/python2.7/site-packages/ansible/executor/playbook_executor.py", line 119, in run
    play.vars[vname] = display.do_var_prompt(vname, private, prompt, encrypt, confirm, salt_size, salt, default)
  File "/usr/lib/python2.7/site-packages/ansible/utils/display.py", line 304, in do_var_prompt
    result = do_prompt(msg, private)
  File "/usr/lib/python2.7/site-packages/ansible/utils/display.py", line 278, in prompt
    return getpass.getpass(msg)
  File "/usr/lib64/python2.7/getpass.py", line 71, in unix_getpass
    passwd = _raw_input(prompt, stream, input=input)
  File "/usr/lib64/python2.7/getpass.py", line 128, in _raw_input
    prompt = str(prompt)
UnicodeEncodeError: 'ascii' codec can't encode character u'\xe9' in position 0: ordinal not in range(128)
```

**Comment by abadger** [comment]
Looks like we can use ```to_native()``` of the prompt.

Python3:
```
>>> getpass.getpass(u'\xe9')
é
'w'
>>> getpass.getpass(b'\xe9')
b'\xe9'
'e'
```

Python2:
```
>>> getpass.getpass(b'é')
é
'd'
>>> getpass.getpass(u'é')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib64/python2.7/getpass.py", line 71, in unix_getpass
    passwd = _raw_input(prompt, stream, input=input)
  File "/usr/lib64/python2.7/getpass.py", line 128, in _raw_input
    prompt = str(prompt)
UnicodeEncodeError: 'ascii' codec can't encode character u'\xe9' in position 0: ordinal not in range(128)
```

So we need a text string on python3 and a byte string on python2... just what ```to_native()``` was designed for.

**Comment by abadger** [comment]
This will be fixed in 2.4.0rc3 and 2.3.3 if we should need to make another 2.3.x release.
